hostinit/setup.go

```diff
@@ -125,15 +139,25 @@ func (init *HostInit) startHosts(ctx context.Context) error {
 		})
 	}
 
-	grip.Info(message.Fields{
-		"GUID":      init.GUID,
-		"runner":    RunnerName,
-		"method":    "startHosts",
-		"num_hosts": started,
-		"runtime":   time.Since(startTime),
-	})
+	m := message.Fields{
+		"GUID":       init.GUID,
+		"runner":     RunnerName,
+		"method":     "startHosts",
+		"num_hosts":  started,
+		"total":      len(hostsToStart),
+		"runtime":    time.Since(startTime),
+		"had_errors": false,
+	}
 
-	return nil
+	if catcher.HasErrors() {
+		m["errors"] = catcher.Resolve()
+		m["had_errors"] = true
+	}
+
+	grip.CriticalWhen(catcher.HasErrors(), m)
+	grip.InfoWhen(!catcher.HasErrors(), m)
+
+	return catcher.Resolve()
```

Comment:
Can catcher.Resolve() be called twice, and have it yield the same result? 

Basically, my concern is that if the catcher has errors, and `m["errors"]` was populated by `catcher.Resolve()`, will calling it again yield an identical error object?

Reply:
yep, it will.